File: backend/TravelPlanService/Models/TravelPlan.cs
namespace TravelPlanService.Models;

public class TravelPlan
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateOnly StartDate { get; set; }
    public DateOnly EndDate { get; set; }
    public decimal Budget { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public List<Destination> Destinations { get; set; } = new();
    public List<Activity> Activities { get; set; } = new();
    public List<ChecklistItem> ChecklistItems { get; set; } = new();
}

File: frontend/src/models/TravelPlan.js
export class TravelPlan {
  constructor(data = {}) {
    this.id = data.id || '';
    this.userId = data.userId || '';
    this.name = data.name || '';
    this.description = data.description || '';
    this.startDate = data.startDate || '';
    this.endDate = data.endDate || '';
    this.budget = data.budget || 0;
    this.notes = data.notes || '';
    this.createdAt = data.createdAt || null;
    this.updatedAt = data.updatedAt || null;
    this.destinations = data.destinations || [];
    this.activities = data.activities || [];
    this.checklistItems = data.checklistItems || [];
  }
}
